src/replication.c

```diff
@@ -739,23 +823,18 @@ create_replica_entry(spec_t *spec, int mgmt_fd)
  */
 replica_t *
 update_replica_entry(spec_t *spec, replica_t *replica, int iofd, char *replicaip, int replica_port) {
-
 	int rc;
+	zvol_io_hdr_t *rio;
 	pthread_t replica_sender_thread;
-	rc = pthread_mutex_init(&replica->r_mtx, NULL); //check
-	if (rc != 0) {
-		REPLICA_ERRLOG("pthread_mutex_init() failed errno:%d\n", errno);
-		return NULL;
-	}
-	rc = pthread_cond_init(&replica->r_cond, NULL); //check
-	if (rc != 0) {
-		REPLICA_ERRLOG("pthread_cond_init() failed errno:%d\n", errno);
-		return NULL;
-	}
+	struct epoll_event event;
+
+	MTX_LOCK(&replica->r_mtx);
```

Comment:
replica's r_mtx can be taken only after spec's rq_mtx.
And, why do we need lock here?